### PR TITLE
TST: resample does not yield empty groups (#10603)

### LINKED ISSUE #10603: Resample yields empty groups
With some parameters, the last group yield by resample is empty. Example : 
```python
import pandas as pd
import numpy as np

df = pd.DataFrame(np.random.normal(size=(10000,4)))
df.index = pd.timedelta_range(start='0s', periods=10000, freq='3906250n')

df.loc['1s':,:].resample('3s',how=lambda x : len(x))
```
Depending of the 'how' function used, this can lead to surprising bugs.

```
INSTALLED VERSIONS
------------------
commit: None
python: 2.7.8.final.0
python-bits: 64
OS: Linux
OS-release: 3.13.0-46-generic
machine: x86_64
processor: x86_64
byteorder: little
LC_ALL: None
LANG: fr_FR.UTF-8

pandas: 0.16.0-294-g45f69cd
nose: 1.3.6
Cython: 0.20.2
numpy: 1.9.2
scipy: 0.14.0
statsmodels: 0.5.0
IPython: 3.0.0-dev
sphinx: 1.2.2
patsy: 0.3.0
dateutil: 2.4.2
pytz: 2015.4
bottleneck: None
tables: 3.1.1
numexpr: 2.4
matplotlib: 1.4.3
openpyxl: 1.7.0
xlrd: 0.9.2
xlwt: 0.7.5
xlsxwriter: None
lxml: None
bs4: 4.3.2
html5lib: 0.999
httplib2: None
apiclient: None
sqlalchemy: 0.8.2
pymysql: None
psycopg2: 2.5.3 (dt dec mx pq3 ext)
```

### PR DESCRIPTION
**Pull request by tkmz-n**
- [x] closes #10603
- [x] tests added / passed
- [x] passes `black pandas`
- [x] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [ ] whatsnew entry

This adds a test to verify that resample does not yield empty groups.

I'm new to contributing to pandas, and I'm not sure where to add this test.
Let me know if this place is inappropriate.

### CODE REVIEW

File: pandas/tests/resample/test_timedelta.py [hunk 1]
```diff
@@ -150,3 +150,15 @@ def test_resample_timedelta_edge_case(start, end, freq, resample_freq):
     tm.assert_index_equal(result.index, expected_index)
     assert result.index.freq == expected_index.freq
     assert not np.isnan(result[-1])
+
+
+def test_resample_timedelta_no_empty_groups():
+    # GH 10603
+    # check that resample does not yield empty groups
+    df = pd.DataFrame(np.random.normal(size=(10000, 4)))
```
Comment (arw2019) [inline pandas/tests/resample/test_timedelta.py:158]: define `df` in a single step: `pd.DataFrame(..., index=...)`

File: pandas/tests/resample/test_timedelta.py [hunk 2]
```diff
@@ -150,3 +150,15 @@ def test_resample_timedelta_edge_case(start, end, freq, resample_freq):
...
+    df.index = pd.timedelta_range(start="0s", periods=10000, freq="3906250n")
+    result = df.loc["1s":, :].resample("3s").apply(lambda x: len(x))
+
+    expected = pd.DataFrame([[768.0] * 4] * 12 + [[528.0] * 4])
```
Comment (arw2019) [inline pandas/tests/resample/test_timedelta.py:162]: same here

File: pandas/tests/resample/test_timedelta.py [hunk 3]
```diff
@@ -150,3 +150,15 @@ def test_resample_timedelta_edge_case(start, end, freq, resample_freq):
     tm.assert_index_equal(result.index, expected_index)
     assert result.index.freq == expected_index.freq
     assert not np.isnan(result[-1])
+
+
+def test_resample_timedelta_no_empty_groups():
+    # GH 10603
+    # check that resample does not yield empty groups
```
Comment (arw2019) [inline pandas/tests/resample/test_timedelta.py:157]: nit-pick: this comment is maybe spurious

File: pandas/tests/resample/test_timedelta.py [hunk 4]
```diff
@@ -150,3 +150,15 @@ def test_resample_timedelta_edge_case(start, end, freq, resample_freq):
     tm.assert_index_equal(result.index, expected_index)
     assert result.index.freq == expected_index.freq
     assert not np.isnan(result[-1])
+
+
+def test_resample_timedelta_no_empty_groups():
```
Comment (arw2019) [inline pandas/tests/resample/test_timedelta.py:155]: nit-pick: `test_resample_with_timedelta_yields_no_empty_groups`

Just to make it easier to tell what the test does if we see it failing

**Review by arw2019 — CHANGES_REQUESTED** [review]
thanks @tkmz-n 

some small comments, o/w lgtm

**Review by arw2019 — APPROVED** [review]

**Comment by alimcmaster1** [comment]
If you merge master CI failures should be resolved

**Pull request merged**